src/main/java/io/aiven/kafka/connect/gcs/config/TimestampSource.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Aiven Kafka GCS Connector
+ * Copyright (c) 2019 Aiven Oy
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.aiven.kafka.connect.gcs.config;
+
+import java.time.Clock;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+
+public interface TimestampSource {
+
+    ZonedDateTime time();
+
+    static TimestampSource of(final Type extractorType) {
+        return of(ZoneOffset.UTC, extractorType);
+    }
+
+    static TimestampSource of(final ZoneId zoneId, final Type extractorType) {
+        switch (extractorType) {
+            case WALLCLOCK:
+                return new WallclockTimestampSource(zoneId);
+            default:
+                throw new IllegalArgumentException(
+                    String.format("Unsupported timestamp extractor type: %s", extractorType)
+                );
+        }
+    }
+
+    enum Type {
+
+        WALLCLOCK;
+
+        public static Type of(final String name) {
+            for (final Type t : Type.values()) {
+                if (t.name().equalsIgnoreCase(name)) {
+                    return t;
+                }
+            }
+            throw new IllegalArgumentException(String.format("Unknown timestamp source: %s", name));
+        }
+
+    }
+
+
+    final class WallclockTimestampSource implements TimestampSource {
+        private final ZoneId zoneId;
+
+        protected WallclockTimestampSource(final ZoneId zoneId) {
+            this.zoneId = zoneId;
+        }
+
+        @Override
+        public ZonedDateTime time() {
+            return ZonedDateTime.ofInstant(Clock.systemUTC().instant(), zoneId);
```

Comment:
`ZonedDateTime.now(zoneId)` ?

Reply:
better `ZonedDateTime.now(Clock.system(zoneId))`

Reply:
It will be done inside, in fact: https://github.com/AdoptOpenJDK/openjdk-jdk9u/blob/master/jdk/src/java.base/share/classes/java/time/ZonedDateTime.java#L215